test/sql-features/mzcompose.py

```diff
@@ -0,0 +1,219 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import argparse
+import os
+import tempfile
+from textwrap import dedent, indent
+
+from materialize.mzcompose import Composition, WorkflowArgumentParser
+from materialize.mzcompose.services import Materialized, Testdrive
+
+SERVICES = [
+    Materialized(unsafe_mode=False),
+    Testdrive(no_reset=True, seed=1),
+]
+
+MZ_SYSTEM_CONNECTION_URL = (
+    "postgres://mz_system:materialize@${testdrive.materialize-internal-sql-addr}"
+)
+USER_CONNECTION_URL = (
+    "postgres://materialize:materialize@${testdrive.materialize-sql-addr}"
+)
+
+
+def header(test_name: str, drop_schema: bool) -> str:
+    """Generate a TD header for a SQL feature test scenario."""
+    header = dedent(
+        f"""
+        # Feature test for SQL feature test: {test_name}
+        #####################################{'#' * len(test_name)}
+        """
+    )
+    # Re-create schema (optional).
+    if drop_schema:
+        header += dedent(
+            f"""
+            > DROP SCHEMA IF EXISTS public CASCADE;
+            > CREATE SCHEMA public /* {test_name} */;
+            """
+        )
+    # Create connections.
+    header += dedent(
+        f"""
+        $ postgres-connect name=user url={USER_CONNECTION_URL}
+        $ postgres-connect name=mz_system url={MZ_SYSTEM_CONNECTION_URL}
+        """
+    )
+    return header.strip()
+
+
+def statement_error(statement: str, error_msg: str) -> str:
+    """Generte a TD command that asserts that `statement` fails with `error_msg`."""
+    return "\n".join(
+        [
+            f"! {indent(statement.strip(), prefix='  ')[2:]}",
+            f"contains:{error_msg}",
+        ]
+    )
+
+
+def statement_ok(statement: str) -> str:
+    """Generte a TD command that executes `statement`."""
+    return f"> {indent(statement.strip(), prefix='  ')[2:]}"
+
+
+def query_ok(query: str) -> str:
+    """Generte a TD command that asserts that a query does not fail."""
+    return "\n".join(
+        [
+            "$ postgres-execute connection=user",
+            query.strip(),
+        ]
+    )
+
+
+def alter_system_set(name: str, value: str) -> str:
+    "Generte a TD command that atlers a system parameter."
+    return dedent(
+        f"""
+        $ postgres-execute connection=mz_system
+        ALTER SYSTEM SET {name} = '{value}';
+        """
+    ).strip()
+
+
+class FeatureTestScenario:
+    """A base class for all feature test scenarios."""
+
+    @classmethod
```

Comment:
The essence of the test logic is in the `phase1` and `phase2` methods and the 
```python
for scenario in scenarios:
   # ...
```
loop in `run_test` below.